server/test/tests.go

```diff
@@ -0,0 +1,256 @@
+		Query          SpanQuery    `json:"query"`
+		ParsedSelector SpanSelector `json:"parsedSelector"`
+	}
+
+	SpanSelector struct {
+		Filters       []SelectorFilter     `json:"filters"`
+		PseudoClass   *SelectorPseudoClass `json:"pseudoClass,omitempty"`
+		ChildSelector *SpanSelector        `json:"childSelector,omitempty"`
+	}
+
+	SelectorFilter struct {
+		Property string `json:"property"`
+		Operator string `json:"operator"`
+		Value    string `json:"value"`
+	}
+
+	SelectorPseudoClass struct {
+		Name     string `json:"name"`
+		Argument *int32 `json:"argument,omitempty"`
+	}
+
+	Summary struct {
+		Runs    int     `json:"runs"`
+		LastRun LastRun `json:"lastRun"`
+	}
+
+	LastRun struct {
+		Time   time.Time `json:"time"`
+		Passes int       `json:"passes"`
+		Fails  int       `json:"fails"`
+	}
+
+	SpanQuery string
+
+	Assertion string
+
+	AssertionExpression struct {
+		LiteralValue LiteralValue
+		Operation    string
+		Expression   *AssertionExpression
+	}
+
+	LiteralValue struct {
+		Value string
+		Type  string
+	}
+
+	RunMetadata map[string]string
+
+	Run struct {
+		ID          int
+		TestID      id.ID
+		TestVersion int
+
+		// Timestamps
+		CreatedAt                 time.Time
+		ServiceTriggeredAt        time.Time
+		ServiceTriggerCompletedAt time.Time
+		ObtainedTraceAt           time.Time
+		CompletedAt               time.Time
+
+		// trigger params
+		State   RunState
+		TraceID trace.TraceID
+		SpanID  trace.SpanID
+
+		// result info
+		TriggerResult trigger.TriggerResult
+		Results       *RunResults
+		Trace         *model.Trace
+		Outputs       maps.Ordered[string, RunOutput]
+		LastError     error
+		Pass          int
+		Fail          int
+
+		Metadata RunMetadata
+
+		// environment
+		Environment environment.Environment
+
+		// transaction
+
+		TransactionID    string
+		TransactionRunID string
+		Linter           model.LinterResult
+	}
+
+	RunResults struct {
+		AllPassed bool
+		Results   maps.Ordered[SpanQuery, []AssertionResult]
+	}
+
+	RunOutput struct {
+		Name     string
+		Value    string
+		SpanID   string
+		Resolved bool
+		Error    error
+	}
+
+	AssertionResult struct {
+		Assertion Assertion
+		AllPassed bool
+		Results   []SpanAssertionResult
+	}
+
+	SpanAssertionResult struct {
+		SpanID        *trace.SpanID
+		ObservedValue string
+		CompareErr    error
+	}
+)
+
+func (t Test) GetID() id.ID {
+	return t.ID
+}
+
+func (t Test) Validate() error {
+	return nil
+}
+
+func (sar SpanAssertionResult) SafeSpanIDString() string {
+	if sar.SpanID == nil {
+		return ""
+	}
+
+	return sar.SpanID.String()
+}
+
+func (t Test) HasID() bool {
+	return t.ID != ""
+}
+
+func (e *AssertionExpression) String() string {
+	if e == nil {
+		return ""
+	}
+
+	if e.Expression == nil {
+		if e.LiteralValue.Type == "attribute" {
+			return fmt.Sprintf("attr:%s", e.LiteralValue.Value)
+		}
+
+		return e.LiteralValue.Value
+	}
+
+	return fmt.Sprintf("%s %s %s", e.LiteralValue.Value, e.Operation, e.Expression.String())
```

Comment:
we could reduce the cyclomatic complexity here

```suggestion
	if e.Expression != nil {
		return fmt.Sprintf("%s %s %s", e.LiteralValue.Value, e.Operation, e.Expression.String())
	}
	
	if e.LiteralValue.Type == "attribute" {
		return fmt.Sprintf("attr:%s", e.LiteralValue.Value)
	}

	return e.LiteralValue.Value
```